kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerState.java

```diff
@@ -335,13 +337,17 @@ synchronized long getEndOffset(String topic, int partition) {
         .map(OffsetAndTimestamp::offset);
   }
 
-  public synchronized boolean expired(long nowMs) {
-    return expiration <= nowMs;
+  public boolean expired(long nowMs) {
+    synchronized (this.expirationLock) {
+      return expiration <= nowMs;
+    }
   }
 
-  public synchronized void updateExpiration() {
-    this.expiration = config.getTime().milliseconds()
-                      + config.getInt(KafkaRestConfig.CONSUMER_INSTANCE_TIMEOUT_MS_CONFIG);
+  public void updateExpiration() {
+    synchronized (this.expirationLock) {
```

Comment:
Resolve a deadlock by making the scope of synchronized small.
(replacing the KafkaConsumerState instance lock with a new KafkaConsumerState::expirationLock instance member lock)